src/server/game/Instances/InstanceSaveMgr.cpp

```diff
@@ -290,6 +290,12 @@ void InstanceSaveManager::LoadResetTimes()
             SetResetTimeFor(mapid, difficulty, t);
             CharacterDatabase.DirectPExecute("INSERT INTO instance_reset VALUES ('%u', '%u', '%u')", mapid, difficulty, (uint32)t);
         }
+        else
+        {
+            // next reset should be in future. If its not, skip to future.
+            while (t < now)
+                t = uint32(((t + MINUTE) / DAY * DAY) + period + diff);
```

Comment:
is `DAY * DAY` meant to do some rounding?

Reply:
Not the `DAY * DAY` but the `/ DAY * DAY`. The first division by DAY leads to the number of days without fractional part. Multiplicating with DAY results in the exact number of seconds for x days.

Reply:
yeah that's what I meant, thanks for clarifying my doubt!